docs/tutorials/workflow/step_by_step.md

```diff
@@ -0,0 +1,744 @@
+
+#### Variables
+Airflow variables should never contain any sensitive information and are used for example for the project_id, bucket
+ names or data location.  
+
+#### Connections
+Airflow connections can contain sensitive information and are often used to store credentials like API keys or
+ usernames and passwords.  
+In the local development environment, the Airflow connections are simply stored in the metastore database.  
+There, the passwords inside the connection configurations are encrypted using Fernet.  
+The value for the Airflow connection should always be a connection URI, see the [Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#generating-a-connection-uri)
+ for more detailed information on how to construct this URI.
+ 
+#### Using a new variable or connection
+To use a new Airflow variable or connection, it has to be added to the relevant class in the airflow_utils file.  
+This file can be found at:  
+`observatory-platform/observatory/platform/utils/airflow_utils.py`
+
+In there are the AirflowVars and AirflowConns classes, these classes make it easier to use the same key name for a
+ variable or connection in many different DAGs.
+The python variable name is used inside the workflow and the value is used inside the `config.yaml` or `config-terraform.yaml`
+ file.
+ 
+For example, to add the airflow variable 'new_variable' and connection 'new_connection', the relevant classes are
+ updated like this:  
+```python
+# Inside observatory-platform/observatory/platform/utils/airflow_utils.py
+class AirflowVars:
+    """Common Airflow Variable names used with the Observatory Platform"""
+
+    # add to existing variables
+    NEW_VARIABLE = "new_variable"
+
+
+class AirflowConns:
+    """Common Airflow Connection names used with the Observatory Platform"""
+
+    # add to existing connections
+    NEW_CONNECTION = "new_connection"
+```
+
+The variable or connection can then be used inside the workflow like this:
+```python
+# Inside my-dags/my_dags/workflows/my_workflow.py
+from observatory.platform.utils.airflow_utils import AirflowVars, AirflowConns
+
+airflow_conn = AirflowConns.NEW_CONNECTION
+airflow_var = AirflowVars.NEW_VARIABLE
+```
+
+The relevant section of both the `config.yaml` and `config-terraform.yaml` files will look like this:
+```yaml
+# User defined Apache Airflow variables:
+airflow_variables:
+  new_variable: my-variable-value
+
+# User defined Apache Airflow Connections:
+airflow_connections:
+  new_connection: http://my-username:my-password@
+```
+
+#### Secrets backend issue, use custom AirflowVariable class
+In the cloud environment that is deployed with terraform, Airflow uses Google Cloud Secret Manager as a secrets
+ backend and both the Airflow variable and connections are stored in there as secrets.
+ 
+Unfortunately, there is currently an issue with Airflow's method to obtain variables when using the secrets backend.  
```

Comment:
This seems to been fixed since 2.0.0a1, it's also related to the other open PR. This part of the documentation will likely be removed as it is not relevant anymore after merging the PR. https://github.com/The-Academic-Observatory/observatory-platform/pull/474